torch/fx/symbolic_trace.py

```diff
@@ -119,7 +120,7 @@ def is_leaf_module(self, m: torch.nn.Module, module_qualified_name : str) -> boo
         """
         return m.__module__.startswith('torch.nn') and not isinstance(m, torch.nn.Sequential)
 
-    def trace(self, root: Union[torch.nn.Module, Callable]) -> Graph:
+    def trace(self, root: Union[torch.nn.Module, Callable], rewrite: bool = True) -> Graph:
```

Comment:
Instead of modifying `trace`, can we just expose a function in `rewriter.py`? I think it would look something like:

```
def rewrite(fn : Union[torch.nn.Module, Callable]) -> Union[torch.nn.Module, Callable]:
  if isinstance(fn, torch.nn.Module):
    # Rewrite this module's forward() and all of its recursive children's
    # forward. Return the new rewritten module hierarchy.
    def rewrite_module(m : torch.nn.Module):
      new_m = copy.copy(m)
      for name, child in new_m.named_children():
          new_m[name] = rewrite_module(child)
      return AST_Rewriter.rewrite(new_m)
    return rewrite_module(fn)
  else:
    # Rewrite this single free function
    return AST_Rewriter.rewrite(fn)
```

We could optionally provide a rewriting Tracer implementation as well:

```
class RewritingTracer(torch.fx.Tracer):
  def trace(self, root: Union[torch.nn.Module, Callable]) -> Graph:
    return super().trace(torch.fx.experimental.rewriter.rewrite(root))
```

I think this all should cleanly decouple this functionality from the Tracer itself